src/Confluent.Kafka/TopicPartition.cs

```diff
@@ -123,5 +125,31 @@ public override int GetHashCode()
         /// </returns>
         public override string ToString()
             => $"{Topic} [{Partition}]";
+
+        /// <summary>
+        ///     Compares the current instance with another object of the same type and returns
+        ///     an integer that indicates whether the current instance precedes, follows, or
+        ///     occurs in the same position in the sort order as the other object.
+        /// </summary>
+        /// <returns>
+        ///     Less than zero:	This instance precedes obj in the sort order.
+        ///     Zero:	This instance occurs in the same position in the sort order as obj.
+        ///     Greater than zero:	This instance follows obj in the sort order.
+        /// </returns>
+        public int CompareTo(object obj)
+        {
+            if (obj.GetType() != this.GetType())
```

Comment:
Is this the desired behaviour if one of the objects is a subclass instance?
Maybe use isInstanceOfType() instead?

Reply:
no one will ever do this, and if they do, i'm fine with an exception happening to stop them, and them complaining, and if it's a legit use-case we'll fix it then.